sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/CatalystConf.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst
+
+import scala.collection.immutable
+
+private[spark] object CatalystConf{
```

Comment:
Thinking about this more, I wonder if it wouldn't be better to keep this very simple

```
trait CatalystConf {
  def caseSensitiveAnalysis: Boolean
}
```

The other functions aren't used, and this way we can keep things like the definition of `"spark.sql.caseSensitive"` to one place in the code base.  The test configurations can be as simple as `case class SimpleConf(caseSensitiveAnalysis: Boolean) extends CatalystConf`.